src/components/intent.ts

```diff
@@ -699,6 +711,32 @@ export class Intent {
         return this.client.getStateEvent(roomId, eventType, stateKey);
     }
 
+    /**
+     * Check if a room is encrypted. If it is, return the algorithm.
+     * @param roomId The room ID to be checked
+     */
+    public async isRoomEncrypted(roomId: string) {
+        if (this.encryptedRooms.has(roomId)) {
+            return this.encryptedRooms.get(roomId);
+        }
+        try {
+            const ev = await this.getStateEvent(roomId, "m.room.encryption");
+            const algo = ev.algorithm as string;
+            if (algo) {
+                this.encryptedRooms.set(roomId, true);
+            }
+            // Return false if missing.
+            return !!algo && algo;
```

Comment:
```suggestion
            const algo = ev.algorithm as unknown;
            if (typeof algo === 'string' && algo) {
                this.encryptedRooms.set(roomId, true);
                return algo;
            }
            // Return false if missing, not a string or empty.
            return false;
```

Reply:
Very good. I don't know what happens to element if you set it to a number :D